src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -112,20 +136,9 @@ HRESULT STDMETHODCALLTYPE CorProfiler::ModuleLoadFinished(ModuleID module_id,
     return S_OK;
   }
 
-  const std::vector<WSTRING> disabled_integration_names =
-      GetEnvironmentValues(environment::disabled_integrations);
-  std::vector<Integration> enabled_integrations =
-      FilterIntegrationsByName(integrations_, disabled_integration_names);
-  if (enabled_integrations.empty()) {
-    // we don't need to instrument anything in this module, skip it
-    Info("CorProfiler::ModuleLoadFinished: ", module_info.assembly.name,
-         ". Skipping (no enabled integrations).");
-    return S_OK;
-  }
-
-  enabled_integrations =
+  std::vector<Integration> filtered_integrations =
       FilterIntegrationsByCaller(integrations_, module_info.assembly.name);
```

Comment:
The original bug was here. Instead of calling
```cpp
enabled_integrations = FilterIntegrationsByCaller(integrations_, module_info.assembly.name);
```
we should have called
```cpp
enabled_integrations = FilterIntegrationsByCaller(enabled_integrations, module_info.assembly.name);
```
because `enabled_integrations` was already filtered by name to remove disabled integrations but `integrations_` was not.